messaging/src/main/java/org/axonframework/messaging/CompletionCallbackMessageStream.java

```diff
@@ -119,4 +109,36 @@ public <R> CompletableFuture<R> reduce(@Nonnull R identity,
                            }
                        });
     }
+
+    static class Single<M extends Message<?>> extends CompletionCallbackMessageStream<M> implements
```

Comment:
Nit: Missing class level JavaDoc. But will add it, as I've also worked on this PR.